src/components/MapPopup/MapPopup.tsx

```diff
@@ -44,13 +46,25 @@ const MapPopup: React.FC<MapPopupProps> = ({
         <Subtitle>{fcMetadata.Area_general}</Subtitle>
       </TopHeader>
       <InfoContent>
-        {areas.map((areaInfo: AreaInfo, index) => (
-          <LineInfo key={index}>
-            <Color color={areaInfo.color} $percent={areaInfo.percent || 45} />
-            {areaInfo.percent && <PercentArea>{areaInfo.percent}%</PercentArea>}
-            <TotalArea>{areaInfo.area}</TotalArea>
-          </LineInfo>
-        ))}
+        {colors.length > 0 ? (
+          areas.map((areaInfo: AreaInfo, index) => (
+            <LineInfo key={index}>
+              <Color color={areaInfo.color} $percent={areaInfo.percent || 45} />
+              {areaInfo.percent && (
+                <PercentArea>{areaInfo.percent}%</PercentArea>
+              )}
+              <TotalArea>{areaInfo.area}</TotalArea>
+            </LineInfo>
+          ))
+        ) : (
+          <List>
+            {areas.map((areaInfo, index) => (
+              <ListItem key={index}>
+                <strong>{areaInfo.area}</strong>
+              </ListItem>
+            ))}
+          </List>
+        )}
```

Comment:
What do you think of moving the outer if statement to the line 52 alone? I think would make the code more readable. The list style could be added by giving the LineInfo component a prop that tell it should appear.

Reply:
i'm not sure i understand it, do you mean like this?: 
```suggestion
{areas.map((areaInfo: AreaInfo, index) => (
          <LineInfo key={index} isColored={colors.length > 0}>
            {colors.length > 0 ? (
              <>
                <Color color={areaInfo.color} $percent={areaInfo.percent || 45} />
                {areaInfo.percent && (
                  <PercentArea>{areaInfo.percent}%</PercentArea>
                )}
                <TotalArea>{areaInfo.area}</TotalArea>
              </>
            ) : (
              <ListItem>
                <strong>{areaInfo.area}</strong>
              </ListItem>
            )}
          </LineInfo>
        ))}
```

Reply:
Not really, I think we could wrap everything with a <List> and every <LineInfo> be a <li> tag and if the colors.length if bigger than 0 we could add the <Color> tag adn if the areaInfo exists we could add the PercentArea